### PR TITLE
Site: Improve documentation for MySQL Specific JSON operators

### PR DESCRIPTION
**Pull request by zhztheplayer**

### CODE REVIEW

File: site/_docs/reference.md [hunk 1]
```diff
@@ -2003,13 +2003,35 @@ Note:
 
 | Operator syntax                                   | Description
 |:------------------------------------------------- |:-----------
-| JSON_TYPE(value)                                  | Returns a string indicating the type of a JSON **value**. This can be an object, an array, or a scalar type
-| JSON_DEPTH(value)                                 | Returns a integer indicating the depth of a JSON **value**. This can be an object, an array, or a scalar type
-| JSON_PRETTY(value)                                | Returns a pretty-printing of JSON **value**.
+| JSON_TYPE(value)                                  | Returns a string value indicating the type of a JSON **value**, *value* is in character string type
+| JSON_DEPTH(value)                                 | Returns a integer value indicating the depth of a JSON **value**, *value* is in character string type
```
Comment (julianhyde) [inline site/_docs/reference.md]: Elsewhere in this doc we define variables and their type, e.g. "In the following: *geom* is a GEOMETRY". You could define *jsonValue* similarly, and therefore you can be more concise when specifying function.

Comment (zhztheplayer) [reply site/_docs/reference.md]: Thank you for taking a look @julianhyde!
I've added type definition for `jsonValue` and `path` right below the parent title "JSON Functions".

**Comment by zhztheplayer** [comment]
Also:
1. Change the format of parameter name from bold to italic;
2. Add doc of unimplemented JSON functions.

**Comment by julianhyde** [comment]
Can you please rebase this? Then I will review.

**Comment by zhztheplayer** [comment]
@julianhyde - rebased onto master.

**Comment by zhztheplayer** [comment]
And I would like to know your thoughts on separating the changes to two commits. Something like:
- Commit 1
Site: Improve documentation for MySQL Specific JSON operators
- Commit 2
Site: Improve documentation for JSON_LENGTH

Then we can cherry-pick Commit 1 to site branch, and Commit 2 should wait for the next release.

**Comment by julianhyde** [comment]
Yes, splitting into two changes would be useful. Thanks for suggesting it.

**Comment by julianhyde** [comment]
Committed to master as 9bce62c6b (corrected in b6d663c0a), and cherry-picked to site as aa025d274 (omitting documentation for `JSON_LENGTH`, which was not implemented as of 1.19).

**Pull request closed without merging**